clippy_lints/src/loops.rs

```diff
@@ -2237,7 +2237,7 @@ struct MutVarsDelegate {
 }
 
 impl<'tcx> MutVarsDelegate {
-    fn update(&mut self, cat: &'tcx Categorization, sp: Span) {
```

Comment:
We need a lint for this. Passing args to a function which are only ever used to recusively call itself